tests/Go/PhpUnit/ClassIsNotWovenConstraint.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/*
+ * Go! AOP framework
+ *
+ * @copyright Copyright 2011, Lisachenko Alexander <lisachenko.it@gmail.com>
+ *
+ * This source file is subject to the license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Go\PhpUnit;
+
+use \PHPUnit_Framework_Constraint as Constraint;
+use \ReflectionClass;
+use Go\Instrument\PathResolver;
+
+/**
+ * Asserts that class is not woven.
+ */
+class ClassIsNotWovenConstraint extends Constraint
+{
+    /**
+     * @var array
+     */
+    private $configuration;
+
+    public function __construct(array $configuration)
+    {
+        parent::__construct();
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function matches($other)
+    {
+        $filename = (new ReflectionClass($other))->getFileName();
```

Comment:
No -> actually it should fail with exception, and it should fail hard! Like, it should blow your computer up.

Constraints here are internal classes (I didn't annotated them with @internal - I didn't noticed that that you are using those) -> and they are not reusable outside AOP framework, on top of that, they should not be used outside `assertThat*`.

So I do not see why we should add extra guards, if it is used wrongly -> it should and it will throw exception.